### PR TITLE
feat: add object-safe-hasownproperty

### PR DESCRIPTION
**Pull request by eljefedelrodeodeljefe**
Hi,

we often want to check just the existence on a deep object and avoid access prototypes or large data. For this purpose it would be cool to consider a .hasOwnProperty util.

Kind regards

### CODE REVIEW

File: packages/object-safe-hasownproperty/index.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  if (!Array.isArray(props)) {
+    throw new Error('props arg must be an array, a string or a symbol');
+  }
+
+  while (props.length) {
+    prop = props.shift();
+
+    // if we are recursing, but met a nullish value, we cannot
+    // access it via .hasOwnProperty and should return negatively
+    if (obj === null || obj === undefined) return false;
```
Comment (angus-c) [inline packages/object-safe-hasownproperty/index.js:50]: a) house style is `if(obj == null)` which does the exact same check
b) house style is {} and new line for `if` statements

File: packages/object-safe-hasownproperty/index.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    if (!Object.prototype.hasOwnProperty.call(obj, prop)) {
+      return false;
+    }
+
+    if (props.length === 0) return true;
```
Comment (angus-c) [inline packages/object-safe-hasownproperty/index.js:56]: house style is {} and new line for `if` statements

Comment (eljefedelrodeodeljefe) [reply packages/object-safe-hasownproperty/index.js]: done. Also removed newlines between blocks (as the ones above where styled)

File: packages/object-safe-hasownproperty/index.js (hunk 1 above)
Comment (eljefedelrodeodeljefe) [reply packages/object-safe-hasownproperty/index.js]: done. Thx.

**Comment by eljefedelrodeodeljefe** [comment]
> This is great, thanks!!
> Please see inline review comments, plus
> a) Can we rename to `just-has`
> b) Can you also append to root level README

@angus-c thanks for the review. I have addressed it accordingly (hopefully) and force pushed. Best regards

**Comment by eljefedelrodeodeljefe** [comment]
> @eljefedelrodeodeljefe thanks!
> Do you have strong objection to just calling it `just-has` (I think it's more memorable, and it seems to be available on npm). I think the `safe` part is implied

@angus-c no objections at all. Whatever you like 💯 . Thanks for the continued guidance. Force pushed. Moved the README part unter just-flip-object, to stay alphabetical. Let me know if you need any other changes.

File: README.md [hunk 3]
```diff
@@ -920,9 +952,9 @@ random([1, 2, 3]); // one of [1, 2, 3], at random
 ```js
 import shuffle from "just-shuffle";
 
-shuffle([1, 2, 3]); 
+shuffle([1, 2, 3]);
 // array with original elements randomly sorted
-shuffle([1, 2, 3], {shuffleAll: true}); 
+shuffle([1, 2, 3], {shuffleAll: true});
```
Comment (eljefedelrodeodeljefe) [inline README.md:957]: @angus-c there was some extraneous whitespace, which vscode autofixed. Do you want me to fix the commit, not to include it?

Comment (angus-c) [reply README.md:957]: that would be great thanks!

File: packages/object-has/package.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "just-has",
+  "version": "0.1.0",
```
Comment (angus-c) [inline packages/object-has/package.json:3]: sorry one more thing. Let's make this 1.0.0

**Comment by eljefedelrodeodeljefe** [comment]
changes are done 👍

**Comment by angus-c** [comment]
Many thanks!

**Comment by angus-c** [comment]
Released! https://anguscroll.com/just/just-has
Do you have a twitter account so I can tweet credit (otherwise I will reference your github account)?

**Pull request merged**